design/datamover.md

```diff
@@ -0,0 +1,208 @@
+   & `DataMoverRestore` CRs. This will help in selecting the data mover implementation during runtime. If the `DataMoverType` name is not defined, then the
+   default `DataMoverType` will be used, which in this case will be `VolSync`
+
+```
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.7.0
+  name: datamovertype.velero.io
+spec:
+  provisioner: <VolSync>
+```
+
+2. DataMoverBackup(DMB): Assuming that the flag to enable DataMover is set to true during Velero install, when a velero backup is created, it triggers DataMover plugin
+   to create the `DataMoverBackup` (DMB) CR in the app namespace. `DataMoverBackup` CR supports either a volume snapshot or a pvc as the type of the backup object. If the
+   velero CSI plugin is used for backup, `VolumeSnapshot` is used as the type or else `PVC` is used.
+
+```
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.7.0
+  name: datamoverbackup.velero.io
+spec:
+  dataMoverType: <datamovertype name> 
+  - type: <VolumeSnapshot|PVC>
+    sourceClaimRef:
+      name: <snapshot_content_name>|<pvc_name>
+      namespace: <namespace>  //optional 
+    config:     //optional based on the datamover impl
+
+```
+
+3. DataMoverRestore(DMR): When a velero restore is triggered, the DataMover plugin looks for `DataMoverBackup` in the backup resources. If it encounters a `DataMoverBackup` resource,
+   then the plugin will create a `DataMoverRestore` CR in the app namespace. It will populate the CR with the details obtained from the `DataMoverBackup` resource.
+
+```
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.7.0
+  name: datamoverrestore.velero.io
+spec:
+  dataMoverType: <datamovertype name>
+  destinationClaimRef:
+    name: <PVC_claim_name>
+    namespace: <namespace>
+  config:     //optional based on the datamover impl
+```
+
+Config section in the above CR is optional. It lets the user specify extra parameters needed by the data mover. For eg: VolSync data mover needs restic secret to perform backup & restore
+
+eg:
+
+```
+apiVersion: velero.io/v1
+kind: DataMoverRestore
+metadata:
+  name: volsync-datamover
+spec:
+  dataMoverType: volsync
+  destinationClaimRef:
+    name: <PVC_claim_name>
+    namespace: <namespace>
+  config:     //optional based on the datamover impl
+    resticSecret:
+      name: <secret_name>
+```
+
+4. DataMover Flag: This will be a Velero install flag, if this is flag is set to true then Velero will be installed with the DataMover Plugin.
+
+5. DataMover Plugin: This is the plugin which will get installed when DataMover Flag is set to true. This plugin's duty would be to create DataMoverBackup(DMB) CR. This
```

Comment:
Does it consist of BackupItemAction and RestoreItemAction plugins?

Reply:
@reasonerjt yes